### PR TITLE
tnix's worst nightmare: down with the legacy (v2)

### PR DESCRIPTION
**Pull request by williamhorning**
client/library compat
- [X] api-client (all versions)
- [X] meo (since 082f1d3)
- [X] roarer-2 (since ce26822)
- [ ] svelte 

changes: 
* _breaking_: kills **all** cloudlink commands except `ping` and `authpswd` (with token only)
* _breaking_: remove `/status` endpoint, use / for `registration_enabled`
* _breaking_: remove timestamp fields except for `e` for new posts
* _feature_: add `page` for all paginated endpoints (to replace `page#`)
* _finally_: update `.gitignore` and readme

### CODE REVIEW

**Comment by mybearworld** [comment]
>  roarer-2 (needs api accounts)

Trying to implement them, but I can't get the endpoint not to respond with a 500 error

**Comment by williamhorning** [comment]
> > roarer-2 (needs api accounts)
> 
> Trying to implement them, but I can't get the endpoint not to respond with a 500 error

that's really odd, is that with this branch or another one? if it's with this one, that's really odd as i haven't touched that code. but if it's with other branches, meo and apic seem to work just fine so try taking a look at their code

**Comment by mybearworld** [comment]
Tnix fixed that bug, I think Roarer 2 is compatible with this now with https://github.com/mybearworld/roarer-2/commit/856cb840979b0f0705d14b6a7be471fa0cea7d2c

**Comment by williamhorning** [comment]
@mybearworld alright cool. lmk when roarer 1 has support please

**Comment by williamhorning** [comment]
the development branch of svelte supports api accounts, so once i cut a release, we should be good

**Comment by delusionsGH** [comment]
the stability of meower svelte shouldn't be compromised so that this pr can be merged, please revert "1.9.0" on the meower-svelte repo, complete 1.9 (and PLEASE TEST IT THIS TIME!) and then merge it back

**Comment by williamhorning** [comment]
since it seems tnix may want to remove cl's dependency on the rest of the backend stuff even without gutting most of cl, this will need a bit more work but that shouldn't be terribly difficult. moving this back to a wip until i do that work

File: rest_api/__init__.py [hunk 1]
```diff
@@ -105,23 +80,6 @@ async def index():
...
-async def get_status():
-    return {
-        "scratchDeprecated": True,
-        "registrationEnabled": app.supporter.registration,
-        "isRepairMode": app.supporter.repair_mode,
-        "ipBlocked": (blocked_ips.search_best(request.ip) is not None),
-        "ipRegistrationBlocked": (registration_blocked_ips.search_best(request.ip) is not None)
-    }, 200
-
-
```
Comment (JoshAtticus) [inline rest_api/__init__.py:124]: Why remove /status, there's no reason to remove it

Comment (tnix100) [reply rest_api/__init__.py:124]: Yup, no reason to remove it.

Comment (williamhorning) [reply rest_api/__init__.py:124]: scratchDeprecated is pointless now, registrationEnabled can be added to / if you care so much, and repair mode / ip blocks give errors for the rest of the api

**Comment by williamhorning** [comment]
in theory the only `cl -> rest` dependency is supporter's post parsing logic. at this point there are still a few `rest -> cl` dependencies, all of which are sending events, getting ulist, or kicking clients (admin api)

File: cloudlink/client.py [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+		self,
+		server,
+		websocket: WebSocketServerProtocol
+	):
+		# Python moment:tm:
+		import cloudlink.server
+		
+		# Setup client
+		self.server: cloudlink.server.CloudlinkServer = server
+		self.websocket = websocket
```
Comment (showierdata9978) [inline cloudlink/client.py:19]: use typing.TYPE_CHECKING to import recursive-type definitions.

https://peps.python.org/pep-0484/#runtime-or-type-checking

**Review by showierdata9978 — CHANGES_REQUESTED** [review]

File: supporter.py [hunk 3]
```diff
@@ -73,7 +73,7 @@ def create_post(
     ) -> tuple[bool, dict]:
         # Create post ID and get timestamp
         post_id = str(uuid.uuid4())
-        ts = timestamp(1).copy()
+        ts = time()
```
Comment (showierdata9978) [inline supporter.py:76]: A small nit; `time.time()` looks better here

Comment (williamhorning) [reply supporter.py:76]: i prefer importing as little as possible whenever possible (even though that doesn't make a difference in python due to its design) and i also prefer explicitly stating what i'm importing from a module whenever possible (no `from x import *` or `import x`

File: rest_api/__init__.py (hunk 1 above)
Comment (JoshAtticus) [reply rest_api/__init__.py]: > scratchDeprecated is pointless now, registrationEnabled can be added to / if you care so much, and repair mode / ip blocks give errors for the rest of the api

womp womp just keep it no need to remove it

**Review by showierdata9978 — APPROVED** [review]

**Review by tnix100 — CHANGES_REQUESTED** [review]
Do not remove /status.

File: rest_api/__init__.py [hunk 4]
```diff
@@ -69,56 +62,40 @@ async def check_auth(headers: TokenHeader):
...
-@app.get("/favicon.ico")  # Favicon, my ass. We need no favicon for an API.
-@hide
-async def favicon_my_ass():
-	return "", 200
-
-
-@app.get("/status")
```
Comment (JoshAtticus) [inline rest_api/__init__.py:114]: Do not remove /status

**Review by JoshAtticus — CHANGES_REQUESTED** [review]

**Comment by tnix100** [comment]
Closing as this has gone stale, CL3 commands have already been removed in develop and main, and there's many unresolved problems.

**Comment by williamhorning** [comment]
> Closing as this has gone stale, CL3 commands have already been removed in develop and main, and there's many unresolved problems.

by many unresolved problems, there’re two: svelte and /status. at this point i don’t really care about /status, so that just leaves svelte. also, the implementations of the server on dev/main still have a lot of stuff that can and should be removed such as the majority of the trusted access logic, the gen_account command, allowing the use of passwords for authpswd, and a fair bit of server code. it’d probably be a pain to handle the conflicts that come from this being stale but i still as though those issues should be addressed somewhere

**Pull request closed without merging**